walkthroughs/howto-outlier-detection/README.md

```diff
@@ -0,0 +1,410 @@
+
+In the App Mesh API, we define Outlier Detection on the server side; that is, the service defines the criteria for its hosts to be considered an outlier. Therefore, Outlier Detection should be defined in the Virtual Node representing the service alongside active health checks. The current implementation of App Mesh's Outlier Detection is based on Envoy's [Outlier Detection](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/outlier#arch-overview-outlier-detection).
+
+Here are the necessary fields to configure Outlier Detection:
+
+- **Interval**: the time between each Outlier Detection sweep. During the sweep is when hosts get ejected or returned to the load balancing set (un-ejected).
+
+- **MaxServerErrors**: the threshold for the number of server errors returned by a given endpoint during an outlier detection interval. If the server error count is greater than or equal to this threshold the host is ejected. A server error is defined as any HTTP 5xx response (or the equivalent for gRPC and TCP connections).
+
+- **BaseEjectionDuration**: The amount of time an outlier host is ejected for is `baseEjectionDuration * number of times this specific host has been ejected`. For example, if baseEjectionDuration is 30 seconds, an outlier host A would first be ejected for 30 seconds and returned to the load balancing set on the next following sweep. If host A later gets ejected again, it will be removed from the load balancing set for `30 seconds * 2 (this host is being ejected the second time) = 1 minute`.
+
+- **MaxEjectionPercent**: The threshold for the max percentage of outlier hosts that can be ejected from the load balancing set. maxEjectionPercent=100 means outlier detection can potentially eject all of the hosts from the upstream service if they are all considered outliers, leaving the load balancing set with zero hosts. In reality, due to a default panic behavior in Envoy, if more than 50% of the endpoints behind a service are considered outliers or are failing health checks, the outlier detection ejection is overturned and traffic will be served to these degraded endpoints. We will cover this in [Panic and Ignore Outlier Detection](https://github.com/rexnp/aws-app-mesh-examples/tree/preview-outlier-detection/walkthroughs/howto-outlier-detection#panic-and-ignore-outlier-detection).
+
+## Step 1: Prerequisites
+
+We will need the latest version of the App Mesh Preview CLI for this walkthrough. You can download and use the latest version using the command below.
+
+```bash
+
+aws configure add-model \
+
+--service-name appmesh-preview \
+
+--service-model https://raw.githubusercontent.com/aws/aws-app-mesh-roadmap/master/appmesh-preview/service-model.json
+
+```
```

Comment:
Can we add commands for both AWS CLI v1 and AWS CLI v2?

Reply:
thanks. that's done now.